### PR TITLE
Add ADR to describe our DB deletion policy

### PR DESCRIPTION
**Pull request by benthorner**
https://trello.com/c/5JMGsuBj/581-follow-up-work-from-versioning-review

### CODE REVIEW

File: docs/adr/000x-restricted-deletion.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+> We may want to delete data in our development environments, but this can also be achieved by resetting the database. We should avoid making changes to the production schema for the convience of development.
+
+## Status
+
+Accepted
+
+## Consequences
+
+Ensure all existing foreign keys are set to `restrict` on deletion.
```
Comment (1pretz1) [inline docs/adr/000x-restricted-deletion.md]: So with this in mind, it means we cant delete a user if they wanted to be removed from the db if they had created documents?

Comment (benthorner) [reply docs/adr/000x-restricted-deletion.md]: Yup - we don't have a concrete use-case for doing that right now. If and when it comes along, then changing these will make more sense, as we'll be able to refer to an example.

File: docs/adr/000x-restricted-deletion.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Different options are appropriate depending on the foreign key.
+
+   * `restrict` is a good choice in a child/parent scenario, where deleting a child should be restricted if it would make the parent invalid. For example, we should not be able to delete an ActiveStorage blob if its parent image still exists.
+   * `nullify` works well if the relationship is optional. For example, our code doesn't currently require a document to have a creator, so it might be reasonable to `creator_id` on document to NULL if we delete a user for some reason.
+   * `cascade` is a good choice in a parent/child scenario, where it makes sense to delete the child when we delete the parent. For example, if we delete a document for some reason, it also makes sense to delete the editions.
+
+## Decision
+
+We do not expect to delete anything from the database. Therefore, we should use `restrict` on all foreign key delete actions. `restrict` is actually the default, but we should continue to specify it explicitly to indicate a conscious decision.
```
Comment (kevindew) [inline docs/adr/000x-restricted-deletion.md]: I'd maybe rephrase "We do not expect to delete anything from the database" to something along the lines of: "We haven't anticipated any real world scenarios that will require us to delete records from the database."

File: docs/adr/000x-restricted-deletion.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+> We may want to delete data in our development environments, but this can also be achieved by resetting the database. We should avoid making changes to the production schema for the convience of development.
```
Comment (kevindew) [inline docs/adr/000x-restricted-deletion.md]: I'd probably put this as a consequence rather than part of the decision and phrase it more that deleting stuff in a development or test environment will be difficult and thus resetting the database is encouraged as an alternative.

**Comment by benthorner** [comment]
@kevindew I've updated to expand on the consequences. Could you take another look?
